### PR TITLE
Add missing `remove_interceptor` API to the Map

### PR DESCRIPTION
**Pull request by mdumandag**
Also, added a test for the `add_interceptor` method.

Protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/399

### CODE REVIEW

File: tests/integration/backward_compatible/proxy/map_test.py [hunk 1]
```diff
@@ -596,6 +617,31 @@ def test_values_with_predicate(self):
     def test_str(self):
         self.assertTrue(str(self.map).startswith("Map"))
 
+    def test_add_interceptor(self):
+        interceptor = MapGetInterceptor("hell")
```
Comment (yuce) [inline tests/integration/backward_compatible/proxy/map_test.py:621]: I don't know why 666 was used as the factory ID in the test jar, but we probably can rename this better :D

File: tests/integration/backward_compatible/proxy/map_test.py [hunk 2]
```diff
@@ -596,6 +617,31 @@ def test_values_with_predicate(self):
...
+        self.assertIsNotNone(registration_id)
+
+        self.map.set(1, "o")
+        value = self.map.get(1)
+        self.assertEqual("hello", value)
+
+    def test_remove_interceptor(self):
+        skip_if_client_version_older_than(self, "5.0")
+
+        interceptor = MapGetInterceptor("hell")
```
Comment (yuce) [inline tests/integration/backward_compatible/proxy/map_test.py:632]: It would be better to use a less-provocative prefix :)

**Review by yuce — APPROVED** [review]
Just two trivial comments.

File: tests/integration/backward_compatible/proxy/map_test.py (hunk 1 above)
Comment (yuce) [reply tests/integration/backward_compatible/proxy/map_test.py:621]: How about `hello`?

Comment (mdumandag) [reply tests/integration/backward_compatible/proxy/map_test.py]: Yeah, probably :smile:

File: tests/integration/backward_compatible/proxy/map_test.py (hunk 2 above)
Comment (mdumandag) [reply tests/integration/backward_compatible/proxy/map_test.py]: This was the first thing that came to my mind when I saw 666 :smiling_imp: 

Anyways, I changed the texts to be nicer

**Pull request merged**